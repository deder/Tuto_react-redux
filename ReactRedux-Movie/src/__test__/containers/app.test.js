import App from '../../containers/app';
import React from 'React';
import videoService from '../../services/__mocks__/video.service';

jest.mock('../../services/video.service.js');

describe("<App />",()=>{
    it('Snapshot', () => {
        const wrapper = mount(
            <App />
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it("toogleShowForm", done => {
        const wrapper = mount(
            <App />
        );
        setTimeout(()=>{
            wrapper.instance().toogleShowForm();
            try {
                expect(wrapper.state().showSearchInput).toEqual(true);
                done();
            } catch (error) {
                done.fail(error)
            }
        })
    });
    
    it("searchHandler", done => {
       const wrapper = mount(
            <App />
        );
        wrapper.instance().searchHandler();
        setTimeout(()=>{
            try {
                wrapper.update();
                const titleFirtRecommendMovie = wrapper.state().recommandMovies[0].title
                expect(titleFirtRecommendMovie).toEqual("The Predator")
                done();
            } catch (error) {
                done.fail(error)
            } 
        })
    });
    it("searchHandler sans vidÃ©o correspondante", done => {
        const wrapper = mount(
             <App />
         );
         wrapper.instance().searchHandler("vide");
         setTimeout(()=>{
             try {
                 wrapper.update();
                 done();
             } catch (error) {
                 done.fail(error)
             } 
         })
     });

        
    it("clickCardHandler", done => {
        const wrapper = mount(
             <App />
         );
         videoService.getPopularMovies().then(({data})=>{
            setTimeout(wrapper.instance().clickCardHandler(data.results[2]))
            setTimeout(()=>{
                try {
                    wrapper.update();
                    const titleCurrentMovieMovie = wrapper.state().currentMovie.title
                    expect(titleCurrentMovieMovie).toEqual("La Nonne")
                    done();
                } catch (error) {
                    done.fail(error)
                } 
            })
        })
     });

     it("clickCardHandler - movie sans youtube key", done => {
        const wrapper = mount(
             <App />
         );
         videoService.getPopularMovies().then(({data})=>{
            setTimeout(wrapper.instance().clickCardHandler(data.results[3]))
            setTimeout(()=>{
                try {
                    wrapper.update();
                    const titleCurrentMovieMovie = wrapper.state().currentMovie.title
                    expect(titleCurrentMovieMovie).toEqual("La Nonne")
                    done();
                } catch (error) {
                    done.fail(error)
                } 
            })
        })
     });

})