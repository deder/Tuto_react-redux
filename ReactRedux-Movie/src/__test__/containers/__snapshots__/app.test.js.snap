// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshot 1`] = `
<div
  className="row"
  style={
    Object {
      "backgroundColor": "#E1E2E1",
    }
  }
>
  <header
    className="col s12"
  >
    <nav
      style={
        Object {
          "backgroundColor": "#004D40",
        }
      }
    >
      <div
        className="nav-wrapper"
      >
        <a
          className="brand-logo"
          href="#"
          style={
            Object {
              "opacity": 0.8,
              "paddingLeft": "15px",
            }
          }
        >
          <i
            className="material-icons"
          >
            subscriptions
          </i>
          My Movies
        </a>
        <ul
          className="right"
          id="nav-mobile"
        >
          <li>
            <a
              className="waves-effect waves-light"
              href="#"
            >
              <i
                className="material-icons"
                onClick={[Function]}
              >
                search
              </i>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
  <div
    className="col l8 m12 s12 offset-l2"
    style={Object {}}
  >
    <div
      className="row"
    >
      <div
        className="col l10 m10 offset-m1 s12"
      />
      <div
        className="col l2 m10 offset-m1 s12"
      />
    </div>
    <div
      className="row hide-on-small-only"
    />
  </div>
  <footer
    className="page-footer"
    style={
      Object {
        "backgroundColor": "#006958",
        "clear": "both",
      }
    }
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col l6 s12"
        >
          <h5
            className="white-text"
          >
            Footer Content
          </h5>
          <p
            className="grey-text text-lighten-4"
          >
            You can use rows and columns here to organize your footer content.
          </p>
        </div>
        <div
          className="col l4 offset-l2 s12"
        >
          <h5
            className="white-text"
          >
            Links
          </h5>
          <ul>
            <li>
              <a
                className="grey-text text-lighten-3"
                href="#!"
              >
                Link 1
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="footer-copyright"
      style={
        Object {
          "backgroundColor": "#004D40",
        }
      }
    >
      <div
        className="container"
      >
        Â© 2014 Copyright Text
        <a
          className="grey-text text-lighten-4 right"
          href="#!"
        >
          More Links
        </a>
      </div>
    </div>
  </footer>
</div>
`;
